### PR TITLE
Add docs for OpenX_JSON serde

### PR DESCRIPTION
**Pull request by mosabua**
## Description

- Also rejig some file format stuff as side effect

## Release notes

(x) This is not user-visible or is docs only, and no release notes are required.

### CODE REVIEW

**Comment by mosabua** [comment]
I still have to do a separate cleanup PR that pull the file formats page apart and into fragments and cleans up the Hive formats .. but for the immediate need this PR should be a reasonable start.

**Comment by 1ambda** [comment]
Thanks for updating hive connector documentation.

I checked that the hive connector doc now includes ‘OPENX_JSON’ Serde so that user can use it to parse JSON object as VARCHAR type.

File: docs/src/main/sphinx/connector/hive.md [hunk 1]
```diff
@@ -34,17 +34,25 @@ The coordinator and all workers must have network access to the Hive metastore
...
+- RCText - RCFile using `ColumnarSerDe`
+- RCBinary - RCFile using `LazyBinaryColumnarSerDe`
 - SequenceFile
-- JSON (using org.apache.hive.hcatalog.data.JsonSerDe)
-- CSV (using org.apache.hadoop.hive.serde2.OpenCSVSerde)
+- JSON - using `org.apache.hive.hcatalog.data.JsonSerDe`
+- CSV - using `org.apache.hadoop.hive.serde2.OpenCSVSerde`
+- OPENX_JSON - OpenX JSON SerDe from `org.openx.data.jsonserde.JsonSerDe`. Find
+  more [details about the Trino implementation in the source repository](https://github.com/trinodb/trino/tree/master/lib/trino-hive-formats/src/main/java/io/trino/hive/formats/line/openxjson/README.md).
+- CSV - using `org.apache.hadoop.hive.serde2.OpenCSVSerde`
```
Comment (dain) [inline docs/src/main/sphinx/connector/hive.md:55]: this is a duplicate

File: docs/src/main/sphinx/object-storage/file-formats.md [hunk 2]
```diff
@@ -108,7 +100,7 @@ with Parquet files performed by supported object storage connectors:
     `parquet_small_file_threshold`.
   - `3MB`
 * - `parquet.experimental.vectorized-decoding.enabled`
-  - Enable using Java Vector API for faster decoding of parquet files. 
+  - Enable using Java Vector API for faster decoding of parquet files.
```
Comment (dain) [inline docs/src/main/sphinx/object-storage/file-formats.md:103]:
```suggestion
  - Enable using Java Vector API (SIMD) for faster decoding of parquet files.
```

**Review by dain — APPROVED** [review]

File: docs/src/main/sphinx/object-storage/file-formats.md (hunk 2 above)
Comment (mosabua) [reply docs/src/main/sphinx/object-storage/file-formats.md:103]: oh .. thats an unrelated formatting only update..

Comment (mosabua) [reply docs/src/main/sphinx/object-storage/file-formats.md:103]: but we can probably add anyway

File: docs/src/main/sphinx/connector/hive.md (hunk 1 above)
Comment (mosabua) [reply docs/src/main/sphinx/connector/hive.md]: reordered and fixed

**Pull request merged**